src/jdk.hotspot.agent/share/classes/sun/jvm/hotspot/SALauncher.java

```diff
@@ -500,6 +500,8 @@ public static void main(String[] args) {
                 func.accept(oldArgs);
             }
         } catch (SAGetoptException e) {
+            System.err.println("SA agent option related exception occured");
```

Comment:
Maybe you could do here:
`System.err.println("SA agent option exception occured: " + e.getMessage);`
and remove the `System.err.println(e.getMessage());` in line 505.